x-pack/plugins/security_solution/public/management/components/paginated_content/paginated_content.tsx

```diff
@@ -0,0 +1,221 @@
+ * support those that BasicTable accept.
+ */
+// eslint-disable-next-line react/display-name
+export const PaginatedContent = memo(
+  <T extends object, C extends ComponentWithAnyProps>({
+    items,
+    ItemComponent,
+    itemComponentProps,
+    itemId,
+    onChange,
+    pagination,
+    loading,
+    noItemsMessage,
+    error,
+    contentClassName,
+    'data-test-subj': dataTestSubj,
+    children,
+  }: PaginatedContentProps<T, C>) => {
+    const [itemKeys] = useState<WeakMap<T, string>>(new WeakMap());
+
+    const getTestId = useTestIdGenerator(dataTestSubj);
+
+    const pageCount = useMemo(
+      () => Math.ceil((pagination?.totalItemCount || 1) / (pagination?.pageSize || 1)),
+      [pagination?.pageSize, pagination?.totalItemCount]
+    );
+
+    const handleItemsPerPageChange: EuiTablePaginationProps['onChangeItemsPerPage'] = useCallback(
+      (pageSize) => {
+        onChange({ pageSize, pageIndex: pagination?.pageIndex || 0 });
+      },
+      [onChange, pagination?.pageIndex]
+    );
+
+    const handlePageChange: EuiTablePaginationProps['onChangePage'] = useCallback(
+      (pageIndex) => {
+        onChange({ pageIndex, pageSize: pagination?.pageSize || 10 });
+      },
+      [onChange, pagination?.pageSize]
+    );
+
+    const generatedBodyItemContent = useMemo(() => {
+      if (error) {
+        return 'string' === typeof error ? <ErrorMessage message={error} /> : error;
+      }
+
+      // This casting here is needed in order to avoid the following a TS error (TS2322)
+      // stating that the attributes given to the `ItemComponent` are not assignable to
+      // type 'LibraryManagedAttributes<C, any>'
+      // @see https://github.com/DefinitelyTyped/DefinitelyTyped/issues/34553
+      const Item = ItemComponent as ComponentType<ReturnType<typeof itemComponentProps>>;
+
+      if (items.length) {
+        return items.map((item) => {
+          let key: Key;
+
+          if (itemId) {
+            key = (item[itemId] as unknown) as Key;
+          } else {
+            if (itemKeys.has(item)) {
+              key = itemKeys.get(item)!;
+            } else {
+              key = generateUUI();
+              itemKeys.set(item, key);
+            }
+          }
+
+          return <Item {...itemComponentProps(item)} key={key} />;
+        });
+      }
+
+      return noItemsMessage || <DefaultNoItemsFound />;
+    }, [ItemComponent, error, itemComponentProps, itemId, itemKeys, items, noItemsMessage]);
+
+    return (
+      <RootContainer data-test-subj={dataTestSubj}>
+        {loading && <EuiProgress size="xs" color="primary" />}
+
+        <div className="body" data-test-subj={getTestId('body')}>
+          <div className={contentClassName}>{children ? children : generatedBodyItemContent}</div>
+        </div>
+
+        {pagination && (
+          <div>
+            <EuiSpacer size="l" />
+
+            <EuiTablePagination
```

Comment:
out of curiosity - Is there an specific reason why you are using this instead of [EuiPagination](https://elastic.github.io/eui/#/navigation/pagination)

Reply:
I just copied it from Trusted apps. I looked at EuiPagination, but it only gives you the pagination portion, not the items per page dropdown. Late yesterday I noticed that we don't have control over the label on the "Pick page Size" dropdown and that the label being used now ("Items per row") is a bit specific to tables and we might want this component to allow control of it.

What do you think? create a custom pagination component for use by this `PaginatedContent` component?

Reply:
Oh, I see... What about a custom component with EuiPagination and a dropdown for page size?

Reply:
Something like this: https://elastic.github.io/eui/#/navigation/pagination#customizable-pagination

Reply:
Yeah, I think that will likely be better and will decouple us from the Pagination component used in Tables.
I will include that change in next PR that will follow this one 
Thanks for the feedback.